letsencrypt/cli.py

```diff
@@ -181,15 +212,88 @@ def run(args, config, plugins):
         return "Configurator could not be determined"
 
     domains = _find_domains(args, installer)
+
+    treat_as_renewal = False
+
+    # Considering the possibility that the requested certificate is
+    # related to an existing certificate.  (config.duplicate, which
+    # is set with --duplicate, skips all of this logic and forces any
+    # kind of certificate to be obtained with treat_as_renewal = False.)
+    if not config.duplicate:
+        identical_names_cert, subset_names_cert = _find_duplicative_certs(
+            domains, config, configuration.RenewerConfiguration(config))
+        # I am not sure whether that correctly reads the systemwide
+        # configuration file.
+        question = None
+        if identical_names_cert:
+            question = (
+                "You have an existing certificate that contains exactly the "
+                "same domains you requested (ref: {0})\n\nDo you want to "
+                "renew and replace this certificate with a newly-issued one?"
+                ).format(identical_names_cert[0])
+        elif subset_names_cert:
+            question = (
+                "You have an existing certificate that contains a portion of "
+                "the domains you requested (ref: {0})\n\nIt contains these "
+                "names: {1}\n\nYou requested these names for the new "
+                "certificate: {2}.\n\nDo you want to replace this existing "
+                "certificate with the new certificate?"
+                ).format(subset_names_cert[0], ", ".join(subset_names_cert[2]),
+                         ", ".join(domains))
+        if question is None:
+            # We aren't in a duplicative-names situation at all, so we don't
+            # have to tell or ask the user anything about this.
+            pass
+        elif zope.component.getUtility(interfaces.IDisplay).yesno(
+                question, "Replace", "Cancel"):
+            treat_as_renewal = True
+        else:
+            reporter_util = zope.component.getUtility(interfaces.IReporter)
+            reporter_util.add_message((
+                "To obtain a new certificate that {0} an existing certificate "
+                "in its domain-name coverage, you must use the --duplicate "
+                "option.\n\nFor example:\n\n{1} --duplicate {2}").format(
+                    "duplicates" if identical_names_cert else "overlaps with",
+                    sys.argv[0], " ".join(sys.argv[1:])),
```

Comment:
Oh sorry, my bad! `main` accepts `cli_args` rather than hard coding `sys.argv` (important for tests) - so it should be probably passed down to the subcommand handler functions. I'll look into this later.